cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageAdmin.java

```diff
@@ -160,8 +199,108 @@ public Lineage computeLineage(final StreamId sourceStream, long startMillis, lon
                                                RunIds.getTime(runId, TimeUnit.MILLISECONDS));
   }
 
-  private Lineage doComputeLineage(final NamespacedEntityId sourceData, long startMillis, long endMillis, int levels)
+
+  private Multimap<RelationKey, Relation> getRollupRelations (Multimap<RelationKey, Relation> relations,
+                                                              Map<ProgramRunId, RunRecordMeta> runRecordMap,
+                                                              Map<String, ProgramRunId> workflowIdMap)
     throws NotFoundException {
+
+    Multimap<RelationKey, Relation> relationsNew = HashMultimap.create();
+    for (Map.Entry<RelationKey, Collection<Relation>> entry : relations.asMap().entrySet()) {
+      for (Relation relation : entry.getValue()) {
+        ProgramRunId workflowProgramRunId = null;
+        String workflowRunId = null;
+        NamespacedEntityId data = null;
+        RunRecordMeta runRecord = runRecordMap.get(
+          new ProgramRunId(relation.getProgram().getNamespace(), relation.getProgram().getApplication(),
+                           relation.getProgram().getType(), relation.getProgram().getProgram(),
+                           relation.getRun().getId()));
+        if (runRecord != null
+          && runRecord.getProperties().containsKey("workflowrunid")) {
```

Comment:
If we create a private method that takes in a Relation and a runRecordMap, and returns a `workflowProgramRunId`, then it makes reading this method easier